src/main/java/frc/team2412/robot/subsystems/IntakeSubsystem.java

```diff
@@ -0,0 +1,146 @@
+package frc.team2412.robot.subsystems;
+
+import static frc.team2412.robot.Hardware.*;
+
+import com.revrobotics.CANSparkBase.IdleMode;
+import com.revrobotics.CANSparkFlex;
+import com.revrobotics.CANSparkLowLevel.MotorType;
+import com.revrobotics.CANSparkMax;
+import edu.wpi.first.networktables.GenericEntry;
+import edu.wpi.first.wpilibj.shuffleboard.Shuffleboard;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import java.util.Map;
+
+public class IntakeSubsystem extends SubsystemBase {
+	// Constants
+	public static final double INTAKE_IN_SPEED = 0.3;
+	public static final double INTAKE_OUT_SPEED = -0.7;
+
+	public static final double INDEX_IN_SPEED = 0.3;
+	public static final double INDEX_OUT_SPEED = -0.3;
+
+	public static final double FEEDER_IN_SPEED = 0.3;
+	public static final double FEEDER_OUT_SPEED = -0.3;
+
+	// Motors
+	private final CANSparkMax intakeMotorFront;
+	private final CANSparkMax intakeMotorBack;
+	private final CANSparkMax intakeMotorLeft;
+	private final CANSparkMax intakeMotorRight;
+
+	private final CANSparkFlex indexMotor;
+
+	private final CANSparkFlex feederMotor;
+
+	// Shuffleboard
+	private final GenericEntry setIntakeSpeedEntry =
+			Shuffleboard.getTab("Intake")
+					.addPersistent("Intake speed - ", INTAKE_IN_SPEED)
+					.withSize(2, 1)
+					.withProperties(Map.of("Min", -1, "Max", 1))
+					.getEntry();
+
+	private final GenericEntry setIndexSpeedEntry =
+			Shuffleboard.getTab("Intake").add("Index speed - ", INDEX_IN_SPEED).withSize(1, 1).getEntry();
+
+	private final GenericEntry setFeederSpeedEntry =
+			Shuffleboard.getTab("Intake")
+					.add("Feeder Speed - ", FEEDER_IN_SPEED)
+					.withSize(1, 1)
+					.getEntry();
+
+	public IntakeSubsystem() {
+		intakeMotorFront = new CANSparkMax(INTAKE_MOTOR_FRONT, MotorType.kBrushless);
+		intakeMotorBack = new CANSparkMax(INTAKE_MOTOR_BACK, MotorType.kBrushless);
+		intakeMotorLeft = new CANSparkMax(INTAKE_MOTOR_LEFT, MotorType.kBrushless);
+		intakeMotorRight = new CANSparkMax(INTAKE_MOTOR_RIGHT, MotorType.kBrushless);
+
+		indexMotor = new CANSparkFlex(INDEX_MOTOR, MotorType.kBrushless);
+
+		feederMotor = new CANSparkFlex(FEEDER_MOTOR, MotorType.kBrushless);
+
+		resetMotors();
+	}
+
+	// commented out since we have two motor types now
+	// private void configureMotor(CANSparkFlex motor) {
+	//	motor.restoreFactoryDefaults();
+	//	motor.setIdleMode(IdleMode.kBrake);
+	//	motor.setSmartCurrentLimit(20);
+	//	motor.burnFlash();
+	// }
```

Comment:
For this we can use [`CANSparkBase`](https://codedocs.revrobotics.com/java/com/revrobotics/cansparkbase) which is the superclass of both `CANSparkFlex` and `CANSparkMax` and pass in the motors as before.

Reply:
that's so sick